src/group/libp2p_ack_stream.erl

```diff
@@ -89,6 +81,23 @@ handle_send(_Kind, From, {Data, Seq}, Timeout, State=#state{}) ->
     Msg = #libp2p_ack_frame_pb{messages=[Data], seqs=[Seq]},
     {ok, {reply, From, pending}, libp2p_ack_stream_pb:encode_msg(Msg), Timeout, State#state{}}.
 
+handle_info(server, {Ref, AcceptResult}, State=#state{reply_ref=Ref}) ->
+    case AcceptResult of
+        {ok, AckRef} ->
+            erlang:demonitor(Ref, [flush]),
+            %% process any queued data in the order it was received
+            NewState = State#state{ack_ref=AckRef, data_queue=[]},
+            [ handle_data(server, Data, NewState) || Data <- lists:reverse(State#state.data_queue) ],
```

Comment:
can we fold over these in case they update the state?

Reply:
They explicitly do not, but I can make it do so for any future-proofing, sure.